cdap-api/src/main/java/co/cask/cdap/api/flow/FlowletConnection.java

```diff
@@ -34,9 +34,15 @@
   private final Type sourceType;
   private final String sourceName;
   private final String targetName;
+  private final String sourceNamespace;
 
   public FlowletConnection(Type sourceType, String sourceName, String targetName) {
+    this(sourceType, null, sourceName, targetName);
+  }
+
+  public FlowletConnection(Type sourceType, String sourceNamespace, String sourceName, String targetName) {
```

Comment:
Will it be better to have a class StreamFlowletConnection which extend this and encapsulated the namesapce since it can only exists if type is stream.